### DISCUSSION TITLE
Keda not scaling as often as it should with cloudwatch metrics

### DISCUSSION THREAD

**Discussion by ricktbaker-montera** [kedacore/keda#4166] [Q&A / Need Help] [answered]
Greetings.  I have keda configured to scale up/down a deployment based off of a cloudwatch metric.   It is scaling up and down, but just not as often I I would expect.   I've never seen it scale less than 10.   It always makes much larger jumps.

Here is my scaled object:

    apiVersion: keda.sh/v1alpha1         
    kind: ScaledObject
    metadata:
      name: model-worker
      namespace: test
    spec:
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: model-worker
      pollingInterval:  30
      cooldownPeriod:   300
        failureThreshold: 60
        replicas: 100
      minReplicaCount: 100
      maxReplicaCount: 400
      advanced:
        restoreToOriginalReplicaCount: false
        horizontalPodAutoscalerConfig:
          name: keda-hpa-model-worker-test
          behavior:
            scaleUp:
              stabilizationWindowSeconds: 0
              policies:
              - type: Percent
                value: 100
                periodSeconds: 15
            scaleDown:
              stabilizationWindowSeconds: 0
              policies:
              - type: Percent
                value: 100
                periodSeconds: 15
      triggers:
        - type: aws-cloudwatch
          metadata:
            namespace: AWS/SQS
            expression: SELECT (SUM("NumberOfMessagesSent") FROM "AWS/SQS" WHERE QueueName = 'myqueue.fifo'
            metricName: NumberOfMessagesSent
            targetMetricValue: '60'
            minMetricValue: '50'
            awsRegion: us-east-1
            identityOwner: operator
            metricCollectionTime: "3600"
            metricStat: "Sum"
            metricStatPeriod: "1800"
            metricEndTimeOffset: "0"

Here is the HPA that gets created:

    Name:                                                               keda-hpa-model-worker-test
    Namespace:                                                          test
    Labels:                                                             app.kubernetes.io/managed-by=keda-operator
                                                                        app.kubernetes.io/name=keda-hpa-model-worker-test
                                                                        app.kubernetes.io/part-of=model-worker
                                                                        app.kubernetes.io/version=2.9.2
                                                                        scaledobject.keda.sh/name=model-worker
    Annotations:                                                        <none>
    CreationTimestamp:                                                  Fri , 27 Jan 2023 17:37:25 -0600
    Reference:                                                          Deployment/model-worker
    Metrics:                                                            ( current / target )
      "s0-aws-cloudwatch-NumberOfMessagesSent" (target average value):  64428m / 60
    Min replicas:                                                       100
    Max replicas:                                                       400
    Behavior:
      Scale Up:
        Stabilization Window: 0 seconds
        Select Policy: Max
        Policies:
          - Type: Percent  Value: 100  Period: 15 seconds
      Scale Down:
        Stabilization Window: 0 seconds
        Select Policy: Max
        Policies:
          - Type: Percent  Value: 100  Period: 15 seconds
     Deployment pods:       124 current / 124 desired
    Conditions:
      Type            Status  Reason              Message
      ----            ------  ------              -------
      AbleToScale     True    ReadyForNewScale    recommended size matches current size
      ScalingActive   True    ValidMetricFound    the HPA was able to successfully calculate a replica count from external metric s0-aws-cloudwatch-NumberOfMessagesSent(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: model-worker,},MatchExpressions:[]LabelSelectorRequirement{},})
      ScalingLimited  False   DesiredWithinRange  the desired count is within the acceptable range
    Events:
      Type    Reason             Age   From                       Message
      ----    ------             ----  ----                       -------
      Normal  SuccessfulRescale  62m   horizontal-pod-autoscaler  New size: 124; reason: All metrics below target
  
So, the above, you can see how it is above the target average and it has been that way for about 30 minutes now, but it still isn't scaling up.   It seems like it only scales up when it is much further above the average.   It is the same for scaling down.   Is there something I am missing here with the acceptable range?   
 
Where it shows the desired count is within acceptable range, that is where I am not understanding.   It is averaging 4 units above my target spread across 124 replicas.  That seems like it should be triggering a scaling event, it's about 500 above in total, so given my target of 60, seems like it should be scaling up by 8 or 9 replicas.

**Comment by tomkerkhove** [comment, marked as answer]
Issue was opened on https://github.com/kedacore/keda/issues/4167.
